### PR TITLE
Created Product controllers and routes

### PR DESCRIPTION
**Pull request by zatkiller**
Created Product routes, Product controllers, added Foreign Key for Product Schema and migration file for Product

### CODE REVIEW

File: src/db/migrations/20201022155502-create-product.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.DataTypes.INTEGER
+      },
+      category: {
+        type: Sequelize.DataTypes.STRING
+      },
+      productname: {
```
Comment (wenhaogoh) [inline src/db/migrations/20201022155502-create-product.js]: Perhaps "productname" should just be "name" according to the naming conventions?

File: src/db/migrations/20201022155502-create-product.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        type: Sequelize.DataTypes.STRING
+      },
+      price: {
+        type: Sequelize.DataTypes.DOUBLE
+      },
+      pic: {
```
Comment (wenhaogoh) [inline src/db/migrations/20201022155502-create-product.js]: Should this be "image" for better clarity?

File: src/models/Product.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    category: {
+      type: DataTypes.STRING,
+    },
+    productname: {
```
Comment (wenhaogoh) [inline src/models/Product.ts]: Same issue here.

File: src/models/Product.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      type: DataTypes.STRING,
+      allowNull: false,
+    },      
+    price: {
+      type: DataTypes.NUMBER,
+    },
+    pic: {
```
Comment (wenhaogoh) [inline src/models/Product.ts]: Same issue here.

File: src/db/migrations/20201022155502-create-product.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        type: Sequelize.DataTypes.STRING
+      },
+      store_id: {
```
Comment (wenhaogoh) [inline src/db/migrations/20201022155502-create-product.js]: Should this column not allow nulls?

File: src/models/Product.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      type: DataTypes.STRING,
+    },
+
```
Comment (wenhaogoh) [inline src/models/Product.ts]: Should this include store_id?

**Review by wenhaogoh — CHANGES_REQUESTED** [review]
Just some nits to pick.

File: docker-compose.yml [hunk 7]
```diff
@@ -1,16 +1,15 @@
-version: "3.8"
```
Comment (ivantjh) [inline docker-compose.yml:1]: Does single and double quotes have a difference on windows? 👀

File: src/db/migrations/20201019135609-create-stores.js [hunk 8]
```diff
@@ -33,6 +33,6 @@ module.exports = {
   },
 
   down: async (queryInterface, Sequelize) => {
-    return queryInterface.dropTable('stores');
+    return queryInterface.dropTable('Stores');
```
Comment (ivantjh) [inline src/db/migrations/20201019135609-create-stores.js:36]: Good catch!

File: src/db/migrations/20201022155502-create-product.js (hunk 1 above)
Comment (ivantjh) [reply src/db/migrations/20201022155502-create-product.js]: Agreed

File: src/db/migrations/20201022155502-create-product.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        type: Sequelize.DataTypes.INTEGER
+      },
+      category: {
+        type: Sequelize.DataTypes.STRING
+      },
+      productname: {
+        type: Sequelize.DataTypes.STRING
+      },
+      price: {
+        type: Sequelize.DataTypes.DOUBLE
```
Comment (ivantjh) [inline src/db/migrations/20201022155502-create-product.js]: Add cannot be null constraint here

File: src/db/migrations/20201022155502-create-product.js (hunk 5 above)
Comment (ivantjh) [reply src/db/migrations/20201022155502-create-product.js]: Change to `storeId`

**Review by ivantjh — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 7 above)
Comment (zatkiller) [reply docker-compose.yml:1]: Idts, but i can't save on double quotes

File: src/models/Product.ts (hunk 6 above)
Comment (zatkiller) [reply src/models/Product.ts]: storeId under the image as well?

File: src/db/migrations/20201022155502-create-product.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.createTable('Products', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.DataTypes.INTEGER,
+      },
+      name: {
```
Comment (wenhaogoh) [inline src/db/migrations/20201022155502-create-product.js:11]: Should the name field be compulsory?

**Review by wenhaogoh — CHANGES_REQUESTED** [review]
LGTM 👍 Just need to make one more change.

**Review by ivantjh — APPROVED** [review]

**Review by wenhaogoh — APPROVED** [review]
LGTM 👍

**Pull request merged**